### DISCUSSION TITLE
Does this project sync external members of GW groups?

### DISCUSSION THREAD

**Discussion by arockett** [slashdevops/idp-scim-sync#46] [Q&A] [answered]
The SSO Sync project fails if there are any external members in a synced group. Does this project

1. Not fail if there are external members, and
2. Correctly sync those external members?

**Comment by christiangda** [comment, marked as answer]
hi @arockett, short answer, no it doesn't.

`idp-scim-sync` works as is documented on the main page, but basically, this program does the following:

### First time

1. Get the groups from `Google Workspace` based on the param `--gws-users-filter` which is the filter for the groups
2. Get `users` that match with the `group's members` get in the previous step
2. __Get `groups` and `users` from `AWS SSO`__
3. Compare the two datasets get in the previous steps, and determine what it neet to do:
> 1.  create (users and groups)
> 2. update (users and groups)
> 3. delete (users and groups)
> 4. equals (users and groups)
4. Execute the previous operations (less the equals) in step 3 in the `AWS SSO` side
5. Store the `state file` with the same information of the `AWS SSO`side

### After the first time

1. Get the groups from `Google Workspace` based on the param `--gws-users-filter` which is the filter for the groups
2. Get `users` that match with the `group's members` get in the previous step
2. __Get `groups` and `users` from `state file`__
3. Compare the two datasets get in the previous steps, and determine what it neet to do:
> 1.  create (users and groups)
> 2. update (users and groups)
> 3. delete (users and groups)
> 4. equals (users and groups)
4. Execute the previous operations (less the equals) in step 3 in the `AWS SSO` side
5. Store the `state file` with the same information of the `AWS SSO`side

if you are planning to use this project I recommend to you read the document [Using SSO](https://github.com/slashdevops/idp-scim-sync/blob/main/docs/Using-SSO.md)
